hq/app/logic/SnykDisplay.scala

```diff
@@ -2,63 +2,60 @@ package logic
 
 import com.fasterxml.jackson.core.JsonParseException
 import play.api.libs.json._
-import utils.attempt.{Attempt, FailedAttempt, Failure}
+import utils.attempt.{Attempt, Failure}
 
 import scala.concurrent.ExecutionContext
 import model._
 import model.Serializers._
 
 object SnykDisplay {
 
-  def getOrganisation(s: String, requiredOrganisation: SnykOrganisationName): Attempt[SnykOrganisation] = {
-    val organisation = for {
-      orglist <- parseJsonToOrganisationList(s)
-      org <- orglist.find(_.name == requiredOrganisation.value)
-    } yield org
-    val error = parseJsonToError(s)
-    val f = Failure(s"Unable to find organisation from $s", s"Could not read Snyk response (${error.error})", 502, None, None)
-    Attempt.fromOption(organisation, FailedAttempt(f))
+  def parseOrganisations(s: String, snykGroupId: SnykGroupId)(implicit ec: ExecutionContext): Attempt[List[SnykOrganisation]] = {
+    for {
+      organisationList <- parseJsonToOrganisationList(s)
+      guardianOrganisationList = organisationList filter {
+        case SnykOrganisation(_, id, Some(group)) => group.id==snykGroupId.value
+        case _ => false
+      }
+    } yield guardianOrganisationList
+
   }
 
   private def parseJsonToError(s: String): SnykError = try {
     Json.parse(s).asOpt[SnykError].getOrElse(SnykError(s))
   } catch {
-    case _: JsonParseException => SnykError(s)
+    case e: JsonParseException => SnykError(e.getMessage)
   }
 
-  private def parseJsonToOrganisationList(s: String): Option[List[SnykOrganisation]] = try {
-    (Json.parse(s) \ "orgs").asOpt[List[SnykOrganisation]]
+  def parseJsonToObject[A](label: String, s: String, f: (String => A)): Attempt[A] = try {
```

Comment:
What is happening here?! You're using exceptions for flow handling. If `f` can fail it should return an `Attempt`.